docs/software-platforms/docker/docker-images.md

```diff
@@ -120,12 +121,107 @@ RUN conda install --channel=conda-forge --name=base conda-lock
+<!-- Should we make one, maybe? -->
+We recommend starting with an R-based image and then installing conda manually.
+In the example below, we are adapting the installation steps used in the [official Miniconda Dockerfile](https://github.com/ContinuumIO/docker-images/blob/main/miniconda3/debian/Dockerfile)
+
+```Dockerfile
+# Dockerfile for an analysis module with both R and conda environments
+# Base image on the Bioconductor 3.19 image
+FROM bioconductor/r-ver:3.19
+
+# set a name for the conda environment
+ARG ENV_NAME=openscpca-analysis
+
+# set environment variables to install miniconda
+ENV PATH="/opt/conda/bin:${PATH}"
+
+# Install Miniconda
+# adapted from https://github.com/ContinuumIO/docker-images/blob/main/miniconda3/debian/Dockerfile
+RUN curl -O https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh \
+  && bash Miniconda3-latest-Linux-x86_64.sh -b -p /opt/conda \
+  && ln -s /opt/conda/etc/profile.d/conda.sh /etc/profile.d/conda.sh  \
+  && echo ". /opt/conda/etc/profile.d/conda.sh" >> ~/.bashrc  \
+  && rm -f Miniconda3-latest-Linux-x86_64.sh \
+  && find /opt/conda/ -follow -type f -name '*.a' -delete \
+  && find /opt/conda/ -follow -type f -name '*.js.map' -delete
+
+# Install conda-lock
+RUN conda install --channel=conda-forge --name=base conda-lock
+
+# Install renv
+RUN Rscript -e "install.packages('renv')"
+
+# Disable the renv cache to install packages directly into the R library
+ENV RENV_CONFIG_CACHE_ENABLED FALSE
+
+# Copy conda lock file to image
+COPY conda-lock.yml conda-lock.yml
+
+# restore from conda-lock.yml file and clean up to reduce image size
+RUN conda-lock install -n ${ENV_NAME} conda-lock.yml && \
   conda clean --all --yes
 
+# Copy the renv.lock file from the host environment to the image
+COPY renv.lock renv.lock
+
+# restore from renv.lock file and clean up to reduce image size
+RUN Rscript -e 'renv::restore()' \
+  && rm -rf ~/.cache/R/renv \
+  && rm -rf /tmp/downloaded_packages \
+  && rm -rf /tmp/Rtmp*
+
 # Activate conda environment on bash launch
 RUN echo "conda activate ${ENV_NAME}" >> ~/.bashrc
 
 # Set entrypoint to bash to activate the environment for any commands
 ENTRYPOINT ["bash", "-l", "-c"]
 ```
+
+!!! tip
+    Note that we install all of Miniconda, `conda-lock` and `renv` before installing any module-specific packages.
+    This is because these steps are not likely to change often, so we can take advantage of Docker's caching to avoid re-installing them every time we build the image.
+
+## Building Docker images
+
+### Local testing
+
+If you have a `Dockerfile` set up for your analysis module, you can build the Docker image locally to test it.
+To match the infrastructure that the OpenScPCA project uses you should use the `docker buildx` system and the `--platform linux/amd64` flag when building the image.
```

Comment:
Will this flag build a _usable_ docker image on all machines though? IIRC, `buildx` indeed makes that happen. Yes?

Reply:
I'm not sure I understand the question? If docker can build it, docker can run it, since building requires running an image.

Reply:
This question was brought to you by OpenPBTA flashbacks of trying to run in containers when I first got my M1 computer. The fact that you don't understand it sounds like things have since resolved themselves and we can move on?